src/agent/src/metrics.rs

```diff
@@ -489,3 +491,82 @@ fn set_gauge_vec_proc_stat(gv: &prometheus::GaugeVec, stat: &procfs::process::St
     gv.with_label_values(&["cutime"]).set(stat.cutime as f64);
     gv.with_label_values(&["cstime"]).set(stat.cstime as f64);
 }
+
+pub struct FsStat {
+    pub bytes_free: u64,
+    pub bytes_used: u64,
+    /// Total file nodes
+    pub files_free: u64,
+    pub files_used: u64,
+    pub status: String,
+}
+
+const OPEN_TIMEOUT_SECS: u64 = 10;
+
+async fn open_mount(path: &str) -> Result<()> {
+    let future = tokio::fs::File::open(path);
+    let duration = Duration::from_secs(OPEN_TIMEOUT_SECS);
+
+    match timeout(duration, future).await {
+        Ok(Ok(_file)) => Ok(()),
+        Ok(Err(err)) => {
+            // No timeout, but failed to open volume
+            Err(err.into())
+        }
+        Err(err) => {
+            // Timeout
+            Err(err.into())
+        }
+    }
+}
+
+pub async fn get_volume_stats(mount: &str) -> Result<FsStat> {
+    let path = Path::new(mount);
+    ensure!(path.exists(), "Mount path not exists");
+
+    let status = open_mount(mount);
+
+    let stat = statfs::statfs(mount)?;
```

Comment:
Question 1: Can use `statfs::statfs`  instead of `open_mount` to check if the status is ok or error?
Question 2: If `statfs::statfs` failed, should return an `internal error`(handled by the caller `get_fs_mount_metrics`) or a normal response by a valid type  `FsStat` struct with a non-ok `status` field ?

Reply:
I'll look at (1).

For (2), I think error makes more sense. @mxpv  WDYT?

Reply:
(2) when `statfs::statfs` fails, we don't have anything useful to return (eg. FsStat will be zeroed), so I'd rather error in this case.